gcloud/translate/client.py

```diff
@@ -0,0 +1,198 @@
+# Copyright 2016 Google Inc. All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Client for interacting with the Google Cloud Translate API."""
+
+
+import httplib2
+import six
+
+from gcloud._helpers import _to_bytes
+from gcloud.translate.connection import Connection
+
+
+ENGLISH_ISO_639 = 'en'
+"""ISO 639-1 language code for English."""
+
+
+class Client(object):
+    """Client to bundle configuration needed for API requests.
+
+    :type key: str
+    :param key: The key used to send with requests as a query
+                parameter.
+
+    :type http: :class:`httplib2.Http` or class that defines ``request()``.
+    :param http: (Optional) HTTP object to make requests. If not
+                 passed, an :class:`httplib.Http` object is created.
+
+    :type target_language: str
+    :param target_language: (Optional) The target language used for
+                            translations and language names. (Defaults to
+                            :data:`ENGLISH_ISO_639`.)
+    """
+
+    def __init__(self, key, http=None, target_language=ENGLISH_ISO_639):
+        self.key = key
+        if http is None:
+            http = httplib2.Http()
+        self.connection = Connection(http=http)
+        self.target_language = target_language
+
+    def get_languages(self, target_language=None):
+        """Get list of supported languages for translation.
+
+        Response
+
+        See: https://cloud.google.com/translate/v2/\
+        discovering-supported-languages-with-rest
+
+        :type target_language: str
+        :param target_language: (Optional) The language used to localize
+                                returned language names. Defaults to the
+                                target language on the current client.
+
+        :rtype: list
+        :returns: List of dictionaries. Each dictionary contains a supported
+                  ISO 639-1 language code (using the dictionary key
+                  ``language``). If ``target_language`` is passed, each
+                  dictionary will also contain the name of each supported
+                  language (localized to the target language).
+        """
+        query_params = {'key': self.key}
+        if target_language is None:
+            target_language = self.target_language
+        if target_language is not None:
```

Comment:
Correct (that's what the unit test that exercises this path does as well). I don't know if it's an "intended" use case but I wanted to allow `target` to remain an optional argument since the `list` API call doesn't require it.